### PR TITLE
324 gb sas

### PR DESCRIPTION
**Pull request by zorge69**
## Pull Request submission

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `environment.yml`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/outputs/output_schemas/gb_sas_schema.toml
Comment (dpurches) [inline src/outputs/output_schemas/gb_sas_schema.toml:1]: Just to check, is the main reason we're using this output schema so that we can rename columns?

Comment (zorge69) [reply src/outputs/output_schemas/gb_sas_schema.toml:1]: To select the columns we want in the output, to put them in correct order and to rename.

File: src/outputs/outputs_helpers.py [hunk 1]
```diff
@@ -34,3 +34,23 @@ def create_output_df(df: pd.DataFrame, output_schema: dict) -> pd.DataFrame:
...
+
+
+def regions() -> dict:
+    """Creates a dictionary of UK regions.
+
+    Args:
+        None
+
+    Returns:
+        (dict): A dictionary of region codes for four UK counties, GB and UK
```
Comment (coatet) [inline src/outputs/outputs_helpers.py]: Think "counties" should be"countries"

**Review by coatet — COMMENTED** [review]
Code runs and looks good. One minor spelling mistake in a docstring that needs correcting.

File: src/outputs/gb_sas.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        df["244"] +
+        df["245"] +
+        df["246"] +
+        df["247"] +
+        df["249"])
+
+    # Create oth_sc
+    df["oth_sc"] = df["242"] + df["248"] + df["250"]
+
+    # Create tau output dataframe with required columns from schema
```
Comment (JenCheshire) [inline src/outputs/gb_sas.py]: should this say gb sas output, rather than tau?

**Review by JenCheshire — COMMENTED** [review]
Looks good to me - just one minor thing flagged in the comments.  Only other observation - entemp and sme appear in the ticket, but say not available in response data, so assume that's why they are not in the output?

File: src/outputs/outputs_helpers.py (hunk 1 above)
Comment (zorge69) [reply src/outputs/outputs_helpers.py]: Of course!

File: src/outputs/gb_sas.py (hunk 2 above)
Comment (zorge69) [reply src/outputs/gb_sas.py]: Just saw it, changed and pushed

**Review by JenCheshire — APPROVED** [review]

**Pull request merged**